src/EnergyPlus/HybridEvapCoolingModel.cc

```diff
@@ -0,0 +1,1591 @@
+				{
+					case SYSTEMOUTPUTS::VENTILATION_AIR_V:  value = po->Supply_Air_Ventilation_Volume; break;
+					case SYSTEMOUTPUTS::SYSTEM_FUEL_USE:  value = po->ElectricalPower; break;
+					case SYSTEMOUTPUTS::SUPPLY_AIR_TEMP:  value = po->SupplyAirTemperature * po->Supply_Air_Mass_Flow_Rate; break;
+					case SYSTEMOUTPUTS::MIXED_AIR_TEMP:  value = po->Mixed_Air_Temperature * po->Supply_Air_Mass_Flow_Rate; break;
+					case SYSTEMOUTPUTS::SUPPLY_MASS_FLOW:  value = po->Supply_Air_Mass_Flow_Rate; break;
+					case SYSTEMOUTPUTS::SUPPLY_AIR_HR:  value = po->SupplyAirW * po->Supply_Air_Mass_Flow_Rate; break;
+					case SYSTEMOUTPUTS::MIXED_AIR_HR:  value = po->Mixed_Air_W * po->Supply_Air_Mass_Flow_Rate; break;
+						
+				}
+				double part_run = po->Runtime_Fraction;
+				averagedVal = averagedVal + value*part_run; 
+				MassFlowDependentDenominator = po->Supply_Air_Mass_Flow_Rate*part_run + MassFlowDependentDenominator;
+			}
+			switch (val)
+			{
+				case SYSTEMOUTPUTS::SUPPLY_AIR_TEMP:  
+					if (MassFlowDependentDenominator==0)
+					{
+						averagedVal = StandbyMode->SupplyAirTemperature;
+					}
+					else averagedVal = averagedVal/ MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::SUPPLY_AIR_HR:  
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->SupplyAirW;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::MIXED_AIR_TEMP:  
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->Mixed_Air_Temperature;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+				case SYSTEMOUTPUTS::MIXED_AIR_HR: 
+					if (MassFlowDependentDenominator == 0)
+					{
+						averagedVal = StandbyMode->Mixed_Air_W;
+					}
+					else averagedVal = averagedVal / MassFlowDependentDenominator; break;
+			}
+			return averagedVal;
+		}
+		double Model::CalculatePartRuntimeFraction(double Mvent, double MinOA_Msa, double RequestedConditioningLoad,double SensibleRoomORZone, double RequestedLatentRoomORZone, double ZoneDehumidificationLoad, double requestedhumidificationload, double humidification)
+		{
+			double PLLatetRatio, PLVentRatio, PLSensibleRatio, PartRuntimeFraction;
+			PLLatetRatio = PLVentRatio = PLSensibleRatio = 0;
+
+			if (Mvent > 0)   PLVentRatio = MinOA_Msa / Mvent;
+			PartRuntimeFraction = 0; //PLVentRatio; tempory fix to deal with fact we haven't yet implimented a ventilation only backup mode.
+			if (SensibleRoomORZone != 0)   PLSensibleRatio = abs(RequestedConditioningLoad) / abs(SensibleRoomORZone);
+			if (PLSensibleRatio > PartRuntimeFraction) PartRuntimeFraction = PLSensibleRatio;
+			if (ZoneDehumidificationLoad > 0)  PLLatetRatio = RequestedLatentRoomORZone / ZoneDehumidificationLoad;
+			if (PLLatetRatio > PartRuntimeFraction) PartRuntimeFraction = PLLatetRatio;
+
+			return PartRuntimeFraction;
+		}
+
+		int Model::SetOperatingSetting(CStepInputs StepIns)
+		{
+
+
```

Comment:
Might as well fix all the extra multiple blank lines in the code as all the other things are getting updated.